### PR TITLE
Add generic JSON-scenario perf client

### PR DESCRIPTION
**Pull request by eddyashton**
We have a user-extensible perf client, but it requires transactions to be generated in C++. Since we expect to support systems where no client code is written in C++ this isn't always appropriate. This PR adds a new perf client which, instead of generating transactions at run-time, simply loads a set of static transactions (another 'scenario') from file.

There's a new test using this client with a simple scenario for the logging app. We don't actually care about the performance result from this, its simply to ensure the new `scenario_perf_client` is regularly built and run.

### CODE REVIEW

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

File: tests/perfclient.py [hunk 1]
```diff
@@ -80,9 +72,11 @@ def add(parser):
...
+    args, unknown_args = cli_args(add, accept_unknown=True)
+
+    unknown_args = [term for arg in unknown_args for term in arg.split(" ")]
 
-    args = cli_args(add)
+    def get_command(*args):
+        return [*args] + unknown_args
 
-    infra.runner.run(args.build_dir, args.package, get_command, args)
+    infra.runner.run(args.build_dir, get_command, args)
```
Comment (jumaffre) [inline tests/perfclient.py]: Was `args.package` unused?

Comment (eddyashton) [reply tests/perfclient.py]: More than unused, it was removed. `runner.run` only takes 3 arguments, so this didn't work. But previously we had no tests calling `perfclient` directly.

**Pull request merged**